server/src/main/java/org/apache/iotdb/db/engine/compaction/no/NoCompactionTsFileManagement.java

```diff
@@ -159,4 +209,25 @@ public void forkCurrentFileList(long timePartition) {
   protected void merge(long timePartition) {
     logger.info("{} no merge logic", storageGroupName);
   }
+
+  private TreeSet<TsFileResource> newSequenceTsFileResources(Long k) {
+    return new TreeSet<>(
+        (o1, o2) -> {
+          try {
+            int rangeCompare =
+                Long.compare(
+                    Long.parseLong(o1.getTsFile().getParentFile().getName()),
+                    Long.parseLong(o2.getTsFile().getParentFile().getName()));
+            return rangeCompare == 0
+                ? compareFileName(o1.getTsFile(), o2.getTsFile())
+                : rangeCompare;
+          } catch (NumberFormatException e) {
+            return compareFileName(o1.getTsFile(), o2.getTsFile());
+          }
+        });
+  }
+
+  private List<TsFileResource> newUnSequenceTsFileResources(Long k) {
+    return new CopyOnWriteArrayList<>();
```

Comment:
Now that you have used `synchronized`, is it necessary to use `CopyOnWriteArrayList` too?